internal/services/dynatrace/dynatrace_tag_rules_resource_test.go

```diff
@@ -0,0 +1,183 @@
+			UserPhoneNumber: os.Getenv("DYNATRACE_USER_PHONE_NUMBER"),
+		},
+	}
+}
+
+func TestAccDynatraceTagRules_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dynatrace_tag_rules", "test")
+	r := NewTagRulesResource()
+	r.preCheck(t)
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccDynatraceTagRules_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dynatrace_tag_rules", "test")
+	r := NewTagRulesResource()
+	r.preCheck(t)
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccDynatraceTagRules_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dynatrace_tag_rules", "test")
+	r := NewTagRulesResource()
+	r.preCheck(t)
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (r TagRulesResource) preCheck(t *testing.T) {
+	if r.dynatraceInfo.UserCountry == "" {
+		t.Skipf("DYNATRACE_USER_COUNTRY must be set for acceptance tests")
+	}
+	if r.dynatraceInfo.UserEmail == "" {
+		t.Skipf("DYNATRACE_USER_EMAIL must be set for acceptance tests")
+	}
+	if r.dynatraceInfo.UserFirstName == "" {
+		t.Skipf("DYNATRACE_USER_FIRST_NAME must be set for acceptance tests")
+	}
+	if r.dynatraceInfo.UserLastName == "" {
+		t.Skipf("DYNATRACE_USER_LAST_NAME must be set for acceptance tests")
+	}
+	if r.dynatraceInfo.UserPhoneNumber == "" {
+		t.Skipf("DYNATRACE_USER_PHONE_NUMBER must be set for acceptance tests")
+	}
+}
+
+func (r TagRulesResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := tagrules.ParseTagRuleID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.Dynatrace.TagRulesClient.Get(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return pointer.To(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+	return pointer.To(true), nil
+}
+
+func (r TagRulesResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_dynatrace_tag_rules" "test" {
+  name       = "acctestragrules%d"
+  monitor_id = azurerm_dynatrace_monitor.test.id
+
+  log_rule {
+    filtering_tag {
+      name   = "Environment"
+      value  = "Prod"
+      action = "Include"
+    }
+  }
+
+  metric_rule {
+    filtering_tag {
+      name   = "Environment"
+      value  = "Prod"
+      action = "Include"
+    }
+  }
```

Comment:
since neither of these are not required, we should be able to create the resource in the basic test without them

Reply:
Will remove.